src/Microsoft.AspNet.Mvc.Razor/RazorTextWriter.cs

```diff
@@ -104,6 +89,20 @@ public override void Write(string value)
         }
 
         /// <inheritdoc />
+        public override void Write(IHtmlContent value)
+        {
+            var htmlTextWriter = TargetWriter as HtmlTextWriter;
+            if (htmlTextWriter == null)
+            {
+                value.WriteTo(TargetWriter, HtmlEncoder);
+            }
+            else
+            {
+                htmlTextWriter.Write(value);
+            }
+        }
```

Comment:
It's abstract there. The implementation is supposed to 'handle' `IHtmlContent`. In this case we handle it by sometimes delegating, and sometimes writing it out.